src/VisualStudio/Core/Def/Implementation/LanguageClient/LanguageServerClient.cs

```diff
@@ -103,38 +97,31 @@ public async Task<Connection> ActivateAsync(CancellationToken cancellationToken)
         }
 
         /// <summary>
-        /// Signals that the extension has been loaded.  The server can be started immediately, or wait for user action to start.  
-        /// To start the server, invoke the <see cref="StartAsync"/> event;
+        /// Signals that the extension has been loaded.
+        /// The caller expects that <see cref="ActivateAsync(CancellationToken)"/> can be called
+        /// immediately following the completion of this method.
         /// </summary>
-        public Task OnLoadedAsync()
+        public async Task OnLoadedAsync()
         {
-            var token = _asyncListener.BeginAsyncOperation("OnLoadedAsync");
+            // initialize things on UI thread
+            await InitializeOnUIAsync().ConfigureAwait(false);
 
-            // set up event stream so that we start LSP server once Roslyn is loaded
-            _eventListener.WorkspaceStarted.ContinueWith(async _ =>
+            // this might get called before solution is fully loaded and before file is opened. 
+            // we delay our OOP start until then, but user might do vsstart before that. so we make sure we start OOP if 
+            // it is not running yet. multiple start is no-op
+            ((RemoteHostClientServiceFactory.RemoteHostClientService)_workspace.Services.GetService<IRemoteHostClientService>()).Enable();
+
+            // wait until remote host is available before let platform know that they can activate our LSP
+            var client = await _workspace.TryGetRemoteHostClientAsync(CancellationToken.None).ConfigureAwait(false);
```

Comment:
Is this used anywhere?

Reply:
Yes, this triggers and waits for the oop to actually load.